CMakeLists.txt

```diff
@@ -1628,24 +1628,33 @@ if(HPX_WITH_APEX)
 
   # We want to track parent dependencies
   hpx_add_config_define(HPX_HAVE_THREAD_PARENT_REFERENCE)
-  # handle APEX library
-  include(GitExternal)
-  git_external(apex
-    https://github.com/khuck/xpress-apex.git
-    ${HPX_WITH_APEX_TAG}
-    ${_hpx_apex_no_update}
-    VERBOSE)
-
-  LIST(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/apex/cmake/Modules")
-  add_subdirectory(apex/src/apex)
-  if(NOT APEX_FOUND)
-    hpx_error("Apex could not be found and HPX_WITH_APEX=On")
+
+  # If APEX_ROOT not specified, local clone into hpx source dir
+  if (NOT APEX_ROOT)
+    # handle APEX library
+    include(GitExternal)
+    git_external(apex
+      https://github.com/khuck/xpress-apex.git
+      ${HPX_WITH_APEX_TAG}
+      ${_hpx_apex_no_update}
+      VERBOSE)
+    if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/apex)
+      set(APEX_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/apex)
+    else()
+      hpx_error("Apex could not be found and HPX_WITH_APEX=On")
+    endif()
+  endif(NOT APEX_ROOT)
+
+  if (NOT APEX_SOURCE_DIR)
+    set(APEX_SOURCE_DIR ${APEX_ROOT}/src/apex)
   endif()
+  LIST(APPEND CMAKE_MODULE_PATH "${APEX_ROOT}/cmake/Modules")
+  include_directories(${APEX_SOURCE_DIR})
+  add_subdirectory(${APEX_SOURCE_DIR} ${CMAKE_BINARY_DIR}/apex/src/apex)
```

Comment:
I know this will work, but I feel a bit uncomfortable defining `APEX_SOURCE_DIR` here. It's actually defined by APEX itself as well and could in principle change once you do `add_subdirectory(${APEX_SOURCE_DIR} ...)`. Maybe you can just do
```
add_subdirectory(${APEX_ROOT}/src/apex ...)
include_directories(${APEX_SOURCE_DIR})
```
?